src/lib/arch/aarch64/Instruction_execute.cc

```diff
@@ -78,8 +78,11 @@ void Instruction::execute() {
   if (isMicroOp_) {
     switch (microOpcode_) {
       case MicroOpcode::LDR_ADDR: {
-        uint16_t regSize =
-            (isScalarData_ || isVectorData_ || isSVEData_) ? 256 : 8;
+        uint16_t regSize = (isInstruction(InsnIdentifier::isScalarDataMask) ||
+                            isInstruction(InsnIdentifier::isVectorDataMask) ||
+                            isInstruction(InsnIdentifier::isSVEDataMask))
+                               ? 256
+                               : 8;
```

Comment:
It was thought that combining the booleans into a single value would tidy up the instruction class and better enforce a "getter-setter" methodology surrounding said instruction identifiers.

I wouldn't agree on reduced readability personally - open for input from the others on this though @jj16791 @JosephMoore25 @ABenC377

Reply:
My preference is for the changes made here as it provides a solution that looks cleaner than a set of boolean getters/setters. I also second Finn's comment regarding "reduced readability". Prior offline analysis has also shown no performance degradation from this change.

Personally, I also find this solution nicer when adding new instruction identifiers than the previous one.

Reply:
I get the point about the use of getters and setters as we previously mixed them with directly accessing the member variable which is not ideal. It seems to me that the a solution of enforcing the previous getters and setters would be better. i.e. we never directly use the 'float_' variable.  Although that does mean we have lots of extra functions in the instruction class class. OR we only use the member variables and don't use getters and setters, they are all private anyway only exposed through the parent class.

Is there a limit on the size of 'isSVEDataMask' etc? And therefor the number of different identifiers we can have? I guess 'int' and therefore 64?

Reply:
On readability it is this

`isScalarData_ || isVectorData_ || isSVEData_
`
vs this

```
isInstruction(InsnIdentifier::isScalarDataMask) ||
                            isInstruction(InsnIdentifier::isVectorDataMask) ||
                            isInstruction(InsnIdentifier::isSVEDataMask))
```

Reply:
I think both approaches are entirely readable and there doesn't seem to be any performance delta, so would plumb for whichever approach is more consistent with our overall methodology.  I'm convinced by the "getter-setter" methodology argument and so vote for changes to be kept.

Reply:
The second one is longer, but I don't think it is any more difficult to understand.  Not sure on our position on verbosity per se is. 

> On readability it is this
> 
> `isScalarData_ || isVectorData_ || isSVEData_ ` vs this
> 
> ```
> isInstruction(InsnIdentifier::isScalarDataMask) ||
>                             isInstruction(InsnIdentifier::isVectorDataMask) ||
>                             isInstruction(InsnIdentifier::isSVEDataMask))
> ```

Reply:
The upper limit for the number of masks would be 64, however with AArch64 currently at 15 masks I do not think we are in danger of reaching this limit for current or future architectures.

If we enforced the getter-setter philosophy with the previous implementation, it would mean for each new identifier we would have to define two new functions. For the new implementation it requires just adding the mask.
(However, given, as Dan said, we previously just accessed the member variables directly in the instruction class this was less of an issue previously.)

Reply:
I've given this a bit of a think. I like the bitfield a lot more than separate bools so lean towards the proposed implementation, but I do agree with Dan where it reduces readability.
I tried to have a think of other solutions, but I think the best option would be stick with this solution and change a couple of things.
- We can OR each mask before passing into isInstruction, this leads to a single call (less syntax) and doesn't require any functionality changes for it to still work
- Remove `DataMask` from the variable names. Given each bit acts as a bool, I think `isVector` for example is descriptive enough, especially when bundled with the Enum name
- Reduce `InsnIdentifier` to `InsnType`. Therefore also `setInstructionIdentifier` to `setInstructionType`
These changes dont change functionality and I believe maintain the same level of clarity, but make some snippets of the code a bit nicer.

It'd lead to the above snippet being:
```
if (!(isInstruction(InsnType::isScalar || 
		    InsnType::isVector ||
		    InsnType::isSVE)) {
      setInstructionType(InsnType::isScalar);
}
```
I think this is the best middleground achievable in this, as I agree with points on both sides.

Reply:
@JosephMoore25 I think these are all good suggestions, especially OR-ing the masks before passing into `isInstruction`, and provide a good balance between new and old. Although, the use of `data` in some of the current masks gives clarity as to what they represent so will just remove the `mask` suffix

Reply:
@tom91136 is correct and that's ruined my morning :( . If we still want my solution, `InsnType` would need to be an `enum` rather than `enum class`, or we could overload the | operator which would be a ~3 line function. Not ideal, but imo still improves clarity (although an argument can be made of how far do you go to improve clarity...)

Also, small mistake in my code snippet where we should be `|`ing the masks, not `||`ing them.

Reply:
A different implementation to the one suggested above was put in place. I'm now satisfied this issue is resolved, but will leave to @dANW34V3R to resolve this comment chain as they raised this point in the first place.